<launch>

  <arg name="initial_pose_x"  default="13.3"/>
  <arg name="initial_pose_y"  default="-0.2"/>
  <arg name="initial_pose_a"  default="3.14"/>


  <node pkg="amcl" type="amcl" name="agv3_amcl">
    <param name="global_frame_id"       value="/map"/>
    <param name="use_map_topic"         value="false"/>

    <!-- scan topic -->
    <remap from="scan"                  to="/agv3/scan"/>
    <remap from="initialpose"           to="/agv3/initialpose"/>
    <remap from="amcl_pose"             to="/agv3/amcl_pose"/>
    <remap from="particlecloud"         to="/agv3/particlecloud"/>

    <param name="odom_frame_id"         value="/agv3/odom"/>
    <param name="base_frame_id"         value="/agv3/base_link"/>
    <param name="odom_model_type"       value="diff"/>
    
    <param name="update_min_d"          value="0.05"/>
    <param name="update_min_a"          value="0.1"/>
    <param name="min_particles"         value="100"/>
    <param name="max_particles"         value="800"/>
    <param name="transform_tolerance"   value="0.3"/>
    <param name="laser_model_type"      value="likelihood_field"/>
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="laser_min_range"       value="-1"/>
    <param name="laser_max_range"       value="-1"/> 
    <param name="laser_max_beams"       value="30"/>
    <param name="laser_z_hit"           value="0.95"/>
    <param name="laser_z_short"         value="0.1"/>
    <param name="laser_z_max"           value="0.05"/>
    <param name="laser_z_rand"          value="0.05"/>
    <param name="laser_sigma_hit"       value="0.2"/> 
    <param name="laser_lambda_short"    value="0.1"/> 

    <param name="resample_interval"     value="1" />
    <param name="recovery_alpha_slow"   value="0.01" />
    <param name="recovery_alpha_fast"   value="0.1" />
    <param name="kld_err"               value="0.05" />
        
    <param name="tf_broadcast"          value="true" />

    <param name="initial_pose_x"        value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y"        value="$(arg initial_pose_y)"/>
    <param name="initial_pose_a"        value="$(arg initial_pose_a)"/>
  </node>

</launch>